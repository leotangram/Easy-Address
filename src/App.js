import { useState } from "react";
import { Button, Container } from "@material-ui/core";

import useStyles from "./AppStyles";
import Form from "./components/form/Form";
import Asentamientos from "./components/asentamientos/Asentamientos";
import CountrySelector from "./components/country-selector/CountrySelector";

function App() {
  const classes = useStyles();

  const [asentamientos, setAsentamientos] = useState([]);
  const [selectedAsentamiento, setSelectedAsentamiento] = useState("");
  const [infoConfirmed, setInfoConfirmed] = useState(false);
  const [country, setCountry] = useState("");
  const [infobyPostal, setInfobyPostal] = useState(false);

  const resetInfo = () => {
    setInfoConfirmed(false);
    setSelectedAsentamiento("");
    setAsentamientos([]);
    setInfobyPostal(false);
    setCountry("");
  };

  return (
    <Container className="App">
      <h1>EASY ADDRESS</h1>
      {!infobyPostal && (
        <CountrySelector country={country} setCountry={setCountry} />
      )}
      {!infoConfirmed && country && (
        <>
          <Form
            country={country}
            infobyPostal={infobyPostal}
            setAsentamientos={setAsentamientos}
            setInfobyPostal={setInfobyPostal}
          />
          {asentamientos && (
            <Asentamientos
              asentamientos={asentamientos}
              setSelectedAsentamiento={setSelectedAsentamiento}
            />
          )}
          {selectedAsentamiento && (
            <Button
              className={classes.confirmButton}
              color="primary"
              fullWidth
              onClick={() => setInfoConfirmed(true)}
              variant="contained"
            >
              Confirma tu informaciÃ³n
            </Button>
          )}
        </>
      )}
      {infoConfirmed && (
        <>
          <h3>Â¡Tu informaciÃ³n ha sido confirmada! ðŸŽ‰ðŸŽ‰ðŸŽ‰ðŸŽ‰</h3>
          <p>Â¿Deseas realizar otra busqueda?</p>
          <Button
            className={classes.confirmButton}
            color="primary"
            onClick={resetInfo}
            variant="contained"
          >
            Realizar otra bÃºsqueda
          </Button>
        </>
      )}
    </Container>
  );
}

export default App;
