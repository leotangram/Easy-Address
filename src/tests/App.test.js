import { shallow } from "enzyme";
import App from "../App";

describe("<Error />", () => {
  const wrapper = shallow(<App />);

  test("should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
