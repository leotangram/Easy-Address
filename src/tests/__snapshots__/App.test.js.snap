// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should show correctly 1`] = `
<WithStyles(ForwardRef(Container))
  className="App"
>
  <h1>
    EASY ADDRESS
  </h1>
  <Form
    setAsentamientos={[Function]}
  />
  <Asentamientos
    asentamientos={Array []}
    setSelectedAsentamiento={[Function]}
  />
</WithStyles(ForwardRef(Container))>
`;
